core/shared/src/main/scala/org/http4s/metrics/CustomLabels.scala

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2013 http4s.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.http4s.metrics
+
+import org.http4s.util.SizedSeq
+import org.http4s.util.SizedSeq0
+
+trait CustomLabels[+SL <: SizedSeq[String]] {
+  def labels: SL
+  def values: SL
+}
```

Comment:
I briefly checked the PR, and the motivation resonates with me 👍🏻 However, I'm a little concerned about adding `SizedSeq`. In the abstract, there is a reference to the Sized type from Shapeless. For our usage, can't we have something simpler, like a `CustomLabels` type holding two List fields: `labels` and `values`, with a set of smart constructors to cover cases with 0, 1, 2, and so on labels (Tuples could be helpful here)?

Reply:
Thanks for reviewing it!
We need `SizedSeq` to enforce that `Metrics` middleware provides a label value list with correct size when it calls `CustomMetricsOps`.
When we define `CustomMetricsOps`, we need a pair of (label name list, default label value list) with the same size, refer the line in the above *Code example*:
```
  .buildCustomMetricsOps(CustomLabels(SizedSeq1("payment_provider"), SizedSeq1("default value1")))
```
But when `Metrics` middleware calls `CustomMetricsOps`, it needs to provide only one list which is the label value list, but with the same size (here we need the `SizedSeq` type), refer this line also in the above *Code example*:
```
    SizedSeq1("Paylly")   # SizedSeq2("Paylly", "Other") would cause compilation error here
```
So we need to abstract out the common size so we can enforce that the size of value list matches what's defined in the corresponding `CustomMetricsOps`.